### PR TITLE
 adding pulsar io docs

### PR DESCRIPTION
**Pull request by jerrypeng**
rehash of PR by @lucperkins

### CODE REVIEW

File: site/_data/cli/pulsar-admin.yaml [hunk 1]
```diff
@@ -510,6 +510,164 @@ commands:
     argument: cluster-name
   - name: delete
     description: Delete namespace isolation policy of a cluster. This operation requires superuser privileges.
+- name: sink
+  description: An interface for managing Pulsar IO sinks (egree data from Pulsar)
```
Comment (srkukarni) [inline site/_data/cli/pulsar-admin.yaml]: egress

File: site/_data/cli/pulsar-admin.yaml [hunk 2]
```diff
@@ -510,6 +510,164 @@ commands:
...
+  subcommands:
+  - name: create
+    description: Create a Pulsar IO sink connector
```
Comment (srkukarni) [inline site/_data/cli/pulsar-admin.yaml]: Here and elsewhere, can we add something that says that this connector will start running in the pulsar cluster.

File: site/_data/cli/pulsar-admin.yaml [hunk 3]
```diff
@@ -510,6 +510,164 @@ commands:
...
+    - flags: --jar
+      description: Path to the Java jar file for the sink
+    - flags: --name
+      description: The sink's name
+    - flags: --namespace
+      description: The sink's namespace
+    - flags: --parallelism
+      description: |
+        "The sink's parallelism factor (i.e. the number of sink instances to run)."
+    - flags: --processingGuarantees
```
Comment (srkukarni) [inline site/_data/cli/pulsar-admin.yaml]: Thinking about it, it looks like processing guarantees make sense for functions, but for sources/sinks we might want to call it different?

File: site/docs/latest/cookbooks/pulsar-io.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+[Pulsar IO](../../getting-started/ConceptsAndArchitecture#pulsar-io) is a feature of Pulsar that enables you to easily create and manage **connectors** that interface with external systems, such as databases and other messaging systems.
+
+## Setup
+
+In order to run Pulsar IO connectors, you'll need to install a source distribution.
```
Comment (srkukarni) [inline site/docs/latest/cookbooks/pulsar-io.md]: why

File: site/_data/cli/pulsar-admin.yaml (hunk 3 above)
Comment (jerrypeng) [reply site/_data/cli/pulsar-admin.yaml]: messagingGuarantees?

Comment (jerrypeng) [reply site/_data/cli/pulsar-admin.yaml]: if we decide to change it, lets to do that as part of a separate PR

**Comment by sijie** [comment]
@srkukarni can you review this again?

**Pull request merged**